drizzlepac/hlautils/catalog_utils.py

```diff
@@ -625,173 +649,169 @@ def transform_list_xy_to_ra_dec(self, list_of_x, list_of_y, drizzled_image):
+            # sources can be detected.
+            threshold = self._nsigma * self.image.bkg_rms_ra
 
-        # get the SCI image data
-        imgarr = self.image.data.copy()
+            # The imgarr should be background subtracted to match the threshold which has no background
+            imgarr_bkgsub = imgarr - self.image.bkg_background_ra
 
-        #
-        # Consider whether the auto-generated kernel (self.image.kernel) would work instead
-        #
-        # Only use a single kernel for now
-        kernel_list = [Gaussian2DKernel, MexicanHat2DKernel]
-        kernel_in_use = kernel_list[0]
-
-        bkg = self.image.bkg
-        threshold = self.image.threshold
-
-        # FIX imgarr should be background subtracted, sextractor uses the filtered_data image
-        imgarr_bkgsub = imgarr - bkg.background
-
-        # *** FIX: should size_source_box size be used in all these places? ***
-        # Create a 2D filter kernel - this will be used to smooth the input
-        # image prior to thresholding in detect_sources().
-        sigma = self.fwhm * gaussian_fwhm_to_sigma
-        kernel = kernel_in_use(sigma, x_size=self.size_source_box, y_size=self.size_source_box)
-        kernel.normalize()
-        if not self.tp_sources:
-            # Source segmentation/extraction
-            # If the threshold includes the background level, then the input image
-            # should NOT be background subtracted.
-            # Note: SExtractor has "connectivity=8" which is the default for this function
-            self.sources = detect_sources(imgarr, threshold, npixels=self.size_source_box, filter_kernel=kernel)
-            self.kernel = kernel  # for use in measure_sources()
-            self.total_source_cat = source_properties(imgarr_bkgsub, self.sources, background=bkg.background,
-                                                      filter_kernel=kernel, wcs=self.image.imgwcs)
+            log.info("SEGMENT. Detecting sources in total image product.")
+            # Note: SExtractor has "connectivity=8" which is the default for detect_sources().
+            self.segm_img = detect_sources(imgarr_bkgsub, threshold, npixels=self._size_source_box, filter_kernel=self.image.kernel)
 
-        # if processing filter product, use sources identified by parent total drizzle product identify_sources() run
-        if self.tp_sources:
+            try:
+                # Deblending is a combination of multi-thresholding and watershed
+                # segmentation. Sextractor uses a multi-thresholding technique.
+                # npixels = number of connected pixels in source
+                # npixels and filter_kernel should match those used by detect_sources()
+                segm_deblended_img = deblend_sources(imgarr_bkgsub, self.segm_img, npixels=self._size_source_box,
+                                                     filter_kernel=self.image.kernel, nlevels=self._nlevels, contrast=self._contrast)
```

Comment:
It may be better to drop 'contrast=...' to the next line...

Reply:
Done.